compiler/verification/conditions.ml

```diff
@@ -142,12 +142,7 @@ let match_and_ignore_outer_reentrant_default (ctx : ctx) (e : typed expr) :
         (Print.expr ()) e)
   | EErrorOnEmpty d ->
     d (* input subscope variables and non-input scope variable *)
-  | _ ->
-    Message.raise_spanned_error (Expr.pos e)
-      "Internal error: this expression does not have the structure expected by \
-       the VC generator:\n\
-       %a"
-      (Print.expr ()) e
+  | _ -> e
```

Comment:
Not sure you want to make that completely silent, maybe a debug message only?

Reply:
This PR should not break the proof mode, this should be updated to follow the new calling conventions.

Reply:
I'll double-check, but I think this is the correct output of this function: if it is wrapped inside `ErrorOnEmpty`, logging calls, etc, we strip them but this way it also works with non-wrapped terms (which may happen with optimisations on, see https://github.com/CatalaLang/catala/pull/528/files/5f5788370f664da0df9a36ea417cd87fefc3641d#diff-1b5c91edc326615f7262dc492bee44c128b18d1f0ae943b3127079c22a98cae7R218-R222

Hm, on second thought it might be `Expr.skip_wrappers e` ?